experimental/dds/tree2/src/domains/leafDomain.ts

```diff
@@ -0,0 +1,45 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { SchemaBuilder } from "../feature-libraries";
+import { ValueSchema } from "../core";
+
+/**
+ * Names in this domain follow https://en.wikipedia.org/wiki/Reverse_domain_name_notation, and are versioned.
+ */
+const builder = new SchemaBuilder("Leaf");
+
+/**
+ * @alpha
+ */
+export const number = builder.leaf("fluidframework.com.leaf.1.number", ValueSchema.Number);
+/**
+ * @alpha
+ */
+export const boolean = builder.leaf("fluidframework.com.leaf.1.boolean", ValueSchema.Boolean);
```

Comment:
I was aiming to keep the version as a version of the leaf library so it would be prat of the prefix we will set in the schema builder in the new API. If we decide we want to also version the leaves themselves, we could go for leaf.2.boolean.1.

That was my thought anyway. I don't think it's a great idea though. If we do want to end up versioning a collection of leaf schema, that would be an API level name and shouldn't go in the type identifier. I'll remove this for now (since I don't think these individual schema will need versioning, but if they do, we can always add .2 or 2 to the end, then stop exporting the old ones them from the leaf module to hide them away in legacy compat code.

Reply:
Makes sense and sounds good to me!